py3status/modules/mpris.py

```diff
@@ -0,0 +1,378 @@
+                })
+            elif button == 'stop':
+                response.append({
+                    'color': '#CCCCCC',
+                    'full_text': ' ' + self.state_stop + ' ',
+                    'index': 'stop',
+                })
+            elif button == 'next':
+                response.append({
+                    'color': '#CCCCCC',
+                    'full_text': ' ' + self.state_next + ' ',
+                    'index': 'next',
+                })
+            elif button == 'previous':
+                response.append({
+                    'color': '#CCCCCC',
+                    'full_text': ' ' + self.state_previous + ' ',
+                    'index': 'previous',
+                })
+
+        return response
+
+    def mpris(self, i3s_output_list, i3s_config):
+        """
+        Get the current output format and return it.
+        """
+        buttons_control = self.buttons_control
+        cached_until = time() + i3s_config['interval']
+        self._dbus = SessionBus()
+
+        running_player = self._detect_running_player()
+        self._player = self._get_player(running_player)
+
+        if self._player is None:
+            buttons_control = None
+            text = self.format_none
+            color = i3s_config['color_bad']
+        else:
+            (text, color) = self._get_text(i3s_config)
+            response_buttons = self._get_response_buttons()
+            if len(response_buttons) == 0:
+                buttons_control = None
+
+        response_text = {
+            'full_text': text,
+            'color': color,
+            'index': 'text'
+        }
+
+        response = {
+            'cached_until': cached_until,
+        }
+
+        if buttons_control == 'left':
+            response['composite'] = response_buttons + [response_text]
+        elif buttons_control == 'right':
+            response['composite'] = [response_text] + response_buttons
+        else:
+            response['composite'] = [response_text]
+
+        return response
+
+    def on_click(self, i3s_output_list, i3s_config, event):
+        """
+        Handles click events
+        """
+        index = event['index']
+        button = event['button']
+
+        if index == 'text':
+            if button == self.button_play:
+                self._player.PlayPause()
+            elif button == self.button_stop:
+                self._player.Stop()
+            elif button == self.button_next:
+                self._player.Next()
+            elif button == self.button_previous:
+                self._player.Previous()
+        else:
+            if button == 1:
+                if index == 'play':
+                    self._player.PlayPause()
+                elif index == 'stop':
+                    self._player.Stop()
+                elif index == 'next':
+                    self._player.Next()
+                elif index == 'previous':
+                    self._player.Previous()
```

Comment:
that `whatever` dict might save a few lines here too :)

or at least make adding new ones really easy